ThirdPartyModules/Magento/InStorePickupShipping.php

```diff
@@ -250,6 +250,67 @@ public function getShipToStoreCarrierMethodCodes(
         return $result;
     }
     
+    /**
+     * @param Magento\InventoryInStorePickupQuote\Model\Address\SetAddressPickupLocation $setAddressPickupLocation
+     * @param Magento\Quote\Model\Quote $quote
+     * @param \stdClass                 $transaction
+     * @return array
+     */
+    public function setInStoreShippingMethodForPrepareQuote(
+        $setAddressPickupLocation,
+        $quote,
+        $transaction
+    ) {
+        try {
+            $shipment = $transaction->order->cart->in_store_shipments[0]->shipment;
+            $referenceCodes = explode('_', $shipment->reference);
+            if ($this->checkIfMagentoInStorePickupByCode($referenceCodes)) {
+                $shippingAddress = $quote->getShippingAddress();
+                $shippingAddress->setCollectShippingRates(true);
+                $pickupLocation = substr_replace($shipment->reference, '', 0, strlen(InStorePickup::DELIVERY_METHOD . '_'));
+                $setAddressPickupLocation->execute($shippingAddress, $pickupLocation);  
+                $shippingAddress->setShippingMethod(InStorePickup::DELIVERY_METHOD)->save();
+            }
+        } catch (\Exception $e) {
+            $this->bugsnagHelper->notifyException($e);
+        }
+    }
+    
+    /**
+     * @param Magento\InventoryInStorePickupQuote\Model\ToQuoteAddress $addressConverter
+     * @param Magento\InventoryInStorePickupApi\Model\GetPickupLocationInterface $getPickupLocation
+     * @param Magento\Quote\Model\Quote $quote
+     * @param \stdClass                 $transaction
+     * @return array
+     */
+    public function setInStoreShippingAddressForPrepareQuote(
+        $addressConverter,
+        $getPickupLocation,
+        $quote,
+        $transaction
+    ) {
+        try {
+            $shipment = $transaction->order->cart->in_store_shipments[0]->shipment;
```

Comment:
can we add isset here?

Reply:
We already check there https://github.com/BoltApp/bolt-magento2/pull/1208/files#diff-7e87dc817d298f1174413fe57820e176d94a88eed33d0d47ea63c974c6d48fb3R516